### PR TITLE
s3: don't double-free path when presign throws after blob creation

### PR DESCRIPTION
**Pull request by robobun**
Fuzzilli found a flaky SIGFPE in the REPRL loop. The minimal repro is:

```js
try { Bun.s3.presign("X"); } catch (e) {}
Bun.gc(true);
```

## What

`Blob.Store.initS3` takes ownership of the `PathLike` it receives: `toThreadSafe()` transfers the underlying `StringImpl` ref for string-backed paths, and for `.encoded_slice` / `.threadsafe_string` the store ends up holding the only handle to the same allocation.

The `S3Client` / `S3File` entry points passed `path` **by value** and kept an `errdefer path.deinit()` active across the subsequent `getPresignUrlFrom()` / `stat()` / `write()` call. When that call threw (e.g. `signRequest` → `error.MissingCredentials`), the store's copy was torn down via `blob.detach()` **and** the caller's `errdefer` freed the same underlying path a second time.

The extra deref under-flows the `StringImpl` refcount. On the next collection `StringImpl::costDuringGC()` does `divideRoundedUp(size, refCount())` with `refCount() == 0` → **SIGFPE**. In debug builds it surfaces earlier as a `reached unreachable code` panic from the throw-scope assertion.

## Fix

Thread `*PathLike` through the `constructS3File*` helpers and neuter the caller's copy (`path.* = .{ .string = bun.PathString.empty }`) once the store has taken ownership. The outer `errdefer path.deinit()` now:
- still cleans up if `getCredentialsWithOptions` throws **before** the store is built, and
- is a no-op on every **post-construction** error path, because the `PathLike` has been emptied.

This is the same neuter-after-transfer pattern `findOrCreateFileFromPath` already uses at `Blob.zig:2119-2121`.

## Test

`test/js/bun/s3/s3-presign-missing-credentials.test.ts` spawns a subprocess with all AWS/S3 env vars cleared, exercises `Bun.s3.presign`, `new S3Client().presign`, `S3Client.presign`, and `Bun.file("s3://…").presign()`, then `Bun.gc(true)`. On current main the subprocess aborts after the first call; with this patch it prints `ERR_S3_MISSING_CREDENTIALS` ×4 and exits 0.

Supersedes #28423 with a narrower change that fixes the ownership bug instead of avoiding the blob construction.

---
Verification (robobun, build #53198 complete): All error-level failures are pre-existing on other PRs and unrelated to S3 path ownership:
- `test-http-should-emit-close-when-connection-is-aborted.ts` (Windows ×3) — fails on 10/10 recent PR builds (#53072–53194).
- `s3-storage-class.test.ts` "writer + options on big file" `S3Error: UnknownError` (macOS 14 aarch64) — pre-existing S3 upload flake in 10/11 recent PR builds; this test exercises multipart upload, not the presign/construct path touched here. Same test passed on retry on macOS 14 x64 in this build.
- `node-http-backpressure-max.test.ts` timeout (macOS 14 x64) — HTTP test, also seen in #53107.
- 2× alpine jobs expired (infra).

The new `s3-presign-missing-credentials.test.ts` passed on every platform that ran tests. Not re-rolling since the HTTP-close failure is 10/10 deterministic across PRs.

### CODE REVIEW

File: src/runtime/webcore/S3Client.zig [hunk 1]
```diff
@@ -154,7 +154,7 @@ pub const S3Client = struct {
         errdefer path.deinit();
 
         const options = args.nextEat();
-        var blob = try S3File.constructS3FileWithS3CredentialsAndOptions(globalThis, path, options, ptr.credentials, ptr.options, ptr.acl, ptr.storage_class, ptr.request_payer);
+        var blob = try S3File.constructS3FileWithS3CredentialsAndOptions(globalThis, &path, options, ptr.credentials, ptr.options, ptr.acl, ptr.storage_class, ptr.request_payer);
         defer blob.detach();
         return S3File.getPresignUrlFrom(&blob, globalThis, options);
```
[the remark this answers is not in the document]
Comment (robobun) [reply src/runtime/webcore/S3Client.zig:159]: Pre-existing — `defer blob.detach()` on these lines dates back to 81fce29fd9a (Jan 2025) and is untouched by this diff. The only change here is `const path` → `var path` / `path` → `&path`. The `content_type` allocation only happens when `options.type` is passed, which none of the crash paths do. Out of scope for the path double-free fix; happy to follow up separately if wanted.
